File: easyboard-frontend-blank/src/comp/api/posting/query/query/PostingQuery.ts
import { fromDomain, CqrsBaseQuery } from '@nara.drama/prologue';
import { Posting } from '../../api-model';


@fromDomain
class PostingQuery extends CqrsBaseQuery<Posting> {
  postingId: string;

  constructor(postingId: string) {
    super(Posting);

    this.postingId = postingId;
  }

  static fromDomain(domain: PostingQuery): PostingQuery {
    const query = new PostingQuery(domain.postingId);

    query.setResponse(domain);
    return query;
  }

  static byId(id: string) {
    return new PostingQuery(id);
  }

}

export default PostingQuery;

File: easyboard-backend_blank/easyboard-aggregate/src/main/java/io/naradrama/easyboard/aggregate/posting/api/query/query/PostingQuery.java
/* 
 COPYRIGHT (c) NEXTREE Inc. 2014
 This software is the proprietary of NEXTREE Inc.
 @since 2014. 6. 10.
*/
package io.naradrama.easyboard.aggregate.posting.api.query.query;

import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;
import io.naradrama.prologue.domain.cqrs.query.CqrsBaseQuery;
import io.naradrama.easyboard.aggregate.posting.domain.entity.Posting;
import io.naradrama.easyboard.aggregate.posting.store.PostingStore;

@Getter
@Setter
@NoArgsConstructor
public class PostingQuery extends CqrsBaseQuery<Posting> {
    /* Autogen by nara studio */
    private String postingId;

    public void execute(PostingStore postingStore) {
        /* Autogen by nara studio */
        setQueryResult(postingStore.retrieve(postingId));
    }
}
